tony-core/src/main/java/com/linkedin/tony/Constants.java

```diff
@@ -137,6 +137,17 @@
   // Metrics
   public static final String METRICS_RPC_PORT = "METRICS_RPC_PORT";
   public static final String MAX_MEMORY_BYTES = "MAX_MEMORY_BYTES";
+  public static final String AVG_MEMORY_BYTES = "AVG_MEMORY_BYTES";
+  // Maximum percent of time one or more kernels was executing on GPU
+  public static final String MAX_GPU_UTILIZATION = "MAX_GPU_UTILIZATION";
+  // Average across GPUs of percent of time one or more kernels was executing on GPU
+  public static final String AVG_GPU_UTILIZATION = "AVG_GPU_UTILIZATION";
+  // Maximum total GPU on-board frame buffer memory used
+  public static final String MAX_GPU_MEMORY_USAGE = "MAX_GPU_MEMORY_USAGE";
+  // Average across GPUs of on-board frame buffer memory used
+  public static final String AVG_GPU_MEMORY_USAGE = "AVG_GPU_MEMORY_USAGE";
```

Comment:
From my understanding, frame buffer memory is in main memory. The GPU itself also has some memory. Can we get metrics for the GPU memory usage?

Reply:
FrameBuffer is in the memory of GPU. There is another metrics from nvidia-smi called "bar1_memory_usage", basically is a map of FB memory so that the memory is visible to CPU. Is bar1 memory the metrics you are referring to?

Reply:
Ah, ok. I got confused from reading https://developer.samsung.com/game/gpu-framebuffer, which has a diagram showing the frame buffer as part of system memory.

I think what you have currently is fine.

Reply:
Added GPU_MAIN_MEMORY_USAGE metrics using "bar1_memory_usage"